### DISCUSSION TITLE
How do i deploy a pocketbase to render?

### DISCUSSION THREAD

**Discussion by USKhokhar** [pocketbase/pocketbase#1973] [Q&A]
I did ask it on Twitter yesterday and pocketbase responded and they told me that render works on disks. So based on that, I'm assuming it is indeed possible to deploy pocketbase server on render. 

But being fairly new to all this backend stuff and working mostly on the client side, i seem to be lost in that. So it would be really helpful if someone could tell me a detailed go through of that process.

My entire project is done im just stuck at the deployment part.

**Comment by ganigeorgiev** [comment]
I cannot provide step-by-step tutorial since I have no experience with Render, but the deployment shouldn't be much different than the flow described in https://pocketbase.io/docs/going-to-production/#minimal-setup.

If you need a sample Dockerfile, you can use the one from https://pocketbase.io/docs/going-to-production/#using-docker.

Please note that the Render's free plan has ephemeral storage, meaning that you have to mount a [Disk](https://render.com/docs/disks), which is a paid service. The idea is that you store your `pb_data` on the mounted disk so that it can persist on vm restart/redeploy/etc.

This is similar to the fly volumes (they offer up to 3gb free disk space), you can check as a reference - https://github.com/pocketbase/pocketbase/discussions/537.

**Reply by USKhokhar** [reply]
is there any free method available as Fly.io has also started to ask for credits now?

**Reply by ganigeorgiev** [reply]
Other users also reported that fly.io started asking for credit card but I wasn't able to reproduce it by creating a new user from their website, not via the cli. See the "Try fly.io for free" link under the card input as shown in https://github.com/pocketbase/pocketbase/discussions/537#discussioncomment-3844703:
![fly_screenshot](https://camo.githubusercontent.com/71b02ab23d365ddcc667462e87778cba4f0ff6c05954c0c5749f56d03ee7ba89/68747470733a2f2f692e696d6775722e636f6d2f5233675367446f2e706e67)
But maybe they have set georestriction based on the client IP, I'm not sure.

Alternative free hosts that other users mentioned (_but I haven't tested them in details and I'm not sure how well they work_):

- [pockethost.io](https://pockethost.io/) by @benallfree
- [Mogenius.com](https://mogenius.com/)
- [Oracle Ampere Free cloud](https://www.oracle.com/cloud/free/) (_this is maybe the most generous but I think also requires a card number_)

**Reply by ptbnam37-creator** [reply]
I tried deploying a pocketbase backend to render, and it worked well
just get the render link and add /_ for the dashboard (idk if added changes to it actually applies)
